ext/oci8/oci8_statement.c

```diff
@@ -1500,6 +1500,10 @@ sb4 php_oci_bind_out_callback(
 
 		/* XXX we assume that zend-zval len has 4 bytes */
 		*alenpp = (ub4*) &Z_STRLEN_P(val);
+		if(sizeof(size_t)==2*sizeof(ub4)){
```

Comment:
A solution using the config.m4 macro PHP_C_BIGENDIAN would be better.

Reply:
Right, this can check `#ifdef WORDS_BIGENDIAN`.

Reply:
@pk1234, I don't know the answer to that question, sorry.  Overall I still feel this `if` test needs to be done at compile time and a macro defined. See how AC_COMPILE_IFELSE is used in other configure file.  Sorry I don't have access to a big endian machine to check this.

Reply:
@cjbj, current code does not work at all on 64bit bigendian machines. So everything that changes the behaviour on bigendian systems only, will be better than what have now.

#ifdef WORDS_BIGENDIAN makes sure that the proposed changes do not influence littleendian systems.

Creating a new macro that will be defined if and only if `sizeof(size_t)==2*sizeof(ub4)` seems to be too much effort to me.

`sizeof(size_t)==2*sizeof(ub4)` can be evaluated at compile time and the optimizer will either do or remove the increment-statement unconditionally. That's exactly the effect of a #ifdef SIZE_T_HAS_TWICE_THE_SIZE_OF_UB4-macro

I therefore feel, that my combination of #ifdef WORDS_BIGENDIAN and if-statement is the best way to go without lots of bigendian/littleendian test-machines.